metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -203,9 +303,63 @@ def _build_metrics_output_node(
                     combine_metrics_join_type=SqlJoinType.INNER,
                 )
                 compute_metrics_nodes.append(
-                    ComputeMetricsNode[SqlDataSetT](
-                        parent_node=compute_metric_node,
-                        metric_specs=[metric_spec],
+                    self.build_computed_metrics_node(
+                        metric_spec=metric_spec,
+                        aggregated_node=compute_metric_node,
+                    )
+                )
+            elif metric.type == MetricType.CONVERSION:
+                metric_input_measure_specs = self._metric_semantics.measures_for_metric(metric_reference)
+                assert len(metric_input_measure_specs) == 2, "Should be base and conversion measure exactly."
+
+                base_measure = next(
+                    filter(
+                        lambda x: metric.type_params.base_measure_reference == x.measure_spec.as_reference,
+                        metric_input_measure_specs,
+                    ),
+                    None,
+                )
+                conversion_measure = next(
+                    filter(
+                        lambda x: metric.type_params.conversion_measure_reference == x.measure_spec.as_reference,
+                        metric_input_measure_specs,
+                    ),
+                    None,
+                )
+                assert (
+                    base_measure
+                ), "Base measure needs to be defined in conversion metrics. Something likely went wrong in validation."
+                assert (
+                    conversion_measure
+                ), "Conversion measure needs to be defined in conversion metrics. Something likely went wrong in validation."
+
+                combined_where = where_constraint
+                if metric.constraint:
+                    metric_constraint = WhereConstraintConverter.convert_to_spec_where_constraint(
+                        self._data_source_semantics, metric.constraint
+                    )
+                    combined_where = combined_where.combine(metric_constraint) if combined_where else metric_constraint
+
+                if metric_spec.constraint:
```

Comment:
Wait, both of these are a thing? Would they ever be different?

Reply:
this is how our current logic works, I didn't actually pay much attention here. Looking at this now I'm not entirely sure why we have both of these. But our current querying logic doesn't actually put the constraint into each MetricSpec during query parsing. So I'm assuming this doesn't get used at all unless somewhere is calling this and explicitly passing in MetricSpecs with constraints defined (which I'm fairly certain no one is doing that). Here's what I'll do, I'm going to remove the `if metric.constraint` block and have that be done in the query parser instead. That way we don't have this messy duplicate here.